packages/api-client/src/odsql/index.ts

```diff
@@ -86,12 +95,18 @@ export class Query {
 
     limit(limit: NumberOrUpdater): Query {
         if (typeof limit === 'number') return this.set('limit', limit.toString());
-        return this.update('limit', (current: string) => limit(Number(current)).toString());
+        return this.update(
+            'limit',
+            limit && ((current: string) => limit(Number(current)).toString())
+        );
     }
 
     offset(offset: NumberOrUpdater): Query {
         if (typeof offset === 'number') return this.set('offset', offset.toString());
-        return this.update('offset', (current: string) => offset(Number(current)).toString());
+        return this.update(
+            'offset',
+            offset && ((current: string) => offset(Number(current)).toString())
+        );
```

Comment:
Note: According to the API team provisional documentation, limit and offset will tend to be removed